Move webhook URL methods to util

vivian opened the pull request:
Remove methods in the client to perform webhook actions by webhook URL.

These methods include:

- delete_webhook_from_url
- execute_webhook_from_url
- update_webhook_with_token_from_url
- update_webhook_from_url

There's not really any *programmatic* reason to have these methods. You don't receive URLs from the API. The only place you get them from is the GUI desktop client, which can easily have the token and ID manually taken out from anyway.

This also removes the `error::UrlError` type and `error::Error::Url` variant, as well as the `url` dependency.

AEnterprise commented:
i disagree, while the gui is the only place you get these, it is where most, if not all, user generated webhooks come from. these might be useful and send to you via command input (use this webhook to log stuff instead of making a new one), used in config files (making the bot self-hostable with a simple config for users who have no clue what a webhook id or token are), ...

while this can be worked around on the bot side, i feel like this is a discord provided and exposed way of representing webhooks that should be supported

7596ff commented:
Based on conversation from discord, moving parsing to a util crate with more url features (such as message links) sounds like an acceptable compromise

vivian commented:
Moved webhook URL parsing to a new `twilight_util::link::webhook` module with a `parse` method. The design of the namespaces is to prepare for additional resource link parsing and formatting to be added to it.

Rebased.

Erk- reviewed (dismissed):
Lgtm

vivian commented:
Rebased
